src/v2/cookbook/clickoutside-directive.md

```diff
@@ -0,0 +1,181 @@
+    const handler = binding.value
+
+    // create a named function for the handler
+    function wrappedHandler(event) {
+      const target = event.target
+      if (!el.contains(target) && el !== target) {
+        handler(event)
+      }
+    }
+
+    // and save it in a property on the element
+    el.__vueClickOutside__ = wrappedHandler
+
+    setTimeout(function() {
+      document.addEventListener('click', wrappedHandler)
+    }, 0)
+  },
+
+  unbind(el) {
+
+    if (el.__vueClickOutside__) {
+      // retrieve the handler from the element
+      const handler = el.__vueClickOutside__
+      // and remove it from the document's click listeners
+      document.removeEventListener('click', handler)
+    }
+
+  }
+})
+```
+
+#### Using a ES6 Map to cache the handler
+
+```JavaScript
+
+// we create the map outside of the directive,
+// because we only need one.
+// We don't have to create one for each instance of our directive.
+var handlerCache = new Map()
+
+Vue.directive('clickoutside', {
+  bind(el, binding) {
+    // ...
+    // el.__vueClickOutside__ = wrappedHandler
+    handlerCache.set(el, wrappedHandler)
+    // ...
+  },
+  unbind(el) {
+
+    if (handlerCache.has(el)) {
+      // get the handler from the Map
+      const hander = handlerCache.get(el)
+      document.removeEventListener('click', handler)
+    }
+    // ...
+  }
+})
+```
+## Argument: Using different event types.
+
+So far, your directive will only listen for `click` events. But there might be situations where we want to listen to other events, like `doubleclick` or `mousedown`.
+
+For this, we can use an argument on our directive. This is what this looks like in our HTML:
+```html
+<modal v-clickoutside:dblclick="handler">
+
+</modal>
+```
+
+To use the argument in our directive, we only have to change the first argument of our `addEventListener()` and `removeEventListener()` calls to use the argument, if present:
+
+```JavaScript
+bind(el, binding) {
+  // ...
+  const eventName = binding.arg || 'click'
+  document.addEventListener(eventName, /*...*/)
+}
+
+unbind(el, binding) {
+  // ...
+  const eventName = binding.arg || 'click'
+  document.removeEventListener(eventName, /*...*/)
+}
+```
+
+## The Final Result
+
+<iframe width="100%" height="500" src="https://jsfiddle.net/Linusborg/g5sh3rcu/embedded/result,html,js,css" allowfullscreen="allowfullscreen" frameborder="0"></iframe>
+<script async src="//"></script>
+
+## Limitations
+
+#### Doesn't work with v-show
+
+`v-show` does not remove the element from the DOM, so `unbind()` is not triggered when `v-show` hides an element. This means that our clickoutside event is still active even if your modal is not visible, and would still fire for each click outside of the modal (which now, is basically *everywhere*)
+
+#### No inline expressions
+
+We can't use an inline expression instead of a handler function. The following will **not** work, because it will be run immediately, not on `click`:
```

Comment:
"...it will be run immediately" sound strange. "...it will ~~be~~ run immediately" or "...it will be **fired** immediately"? Not sure though.